pkg/api/types.go

```diff
@@ -1410,6 +1412,55 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// DisruptionBudget is an object to define the max disruption that can be caused to a collection of pods
+type DisruptionBudget struct {
```

Comment:
My preference would be to make this an API resource and move it to extensions. That will reduce risk when introducing such a major new mechanism.

We'd need to add a selector to identify the pods to which it applied.

That would make it easier to orphan and adopt pods, using the same mechanism as controllers and services.

I think it is ok and probably even desirable for multiple disruption budgets to be applied to a pod simultaneously, such as when a pod is a member of a canary or shard and also a member of a whole application tier.